### DISCUSSION TITLE
2023.3.2b1

### DISCUSSION THREAD

**Discussion by Djelibeybi** [Djelibeybi/hass-soma-connect#22] [Announcements]
## What's Changed
* chore: add debug logging to help troubleshoot SOMA Connect by @Djelibeybi in https://github.com/Djelibeybi/hass-soma-connect/pull/21

**Full Changelog**: https://github.com/Djelibeybi/hass-soma-connect/compare/2023.3.1...2023.3.2b1

<hr /><em>This discussion was created from the release <a href='https://github.com/Djelibeybi/hass-soma-connect/releases/tag/2023.3.2b1'>2023.3.2b1</a>.</em>

**Comment by ziptbm** [comment]
@Djelibeybi I've configured this beta build, logger, and restarted as prescribed in this [thread](https://community.home-assistant.io/t/soma-smart-shades-2/202224/135), but I do need a pointer on how to pull the logs for this.  I've enabled debugging on the integration, reproduced the situation, but I'm not seeing anything in the logs via the HA UI.   How should I get you this information from my install?

**Reply by Djelibeybi** [reply]
On the logs screen of Home Assistant, click "Download full logs" and that file should have the debug content.

**Reply by Djelibeybi** [reply]
Feel free to email me the logs at git@dje.li if you don't want to upload them to GitHub. They can sometimes contain details that shouldn't really be public.

**Reply by ziptbm** [reply]
> On the logs screen of Home Assistant, click "Download full logs" and that file should have the debug content.

I did, but didn’t find anything with a few quick searches.  What am I looking for?  I’d prefer to provide just that info and not the full logs.

**Reply by Djelibeybi** [reply]
All lines that include `custom_components.soma_connect` should be included. If you don't have anything with that component, then it's not being used.

**Reply by ziptbm** [reply]
I’m not doing something right as there is nothing in the log with that.  I’ve even redownloaded the integration from hacs and rebooted, but nothing.  It clearly shows that it’s a custom component in the UI and debug logging is enabled.

**Reply by Djelibeybi** [reply]
Have you enabled "Show beta versions" in HACS and then selected `2023.3.2b1` as the version to download? You may have to click "Update information" in the three-dots menu in HACS to update the list of available versions.

**Reply by Djelibeybi** [reply]
Also make sure that the SOMA Connect block on your Devices screen has the "Provided by custom integration" icon and that the documentation links point here. I notice you can enable/disable debug logging via that screen too. The frontend developers keep adding buttons I never notice. :)

**Reply by ziptbm** [reply]
Yes, 2023.3.2b1 shows installed.  
![52D51932-9D7A-4C67-BB7B-A7BAA1494712](https://user-images.githubusercontent.com/46136563/224445889-6591ba60-8f0a-493b-bca6-333dd0cd3ebf.png)


I’ll next try uninstalling and reinstalling to see if that fixes it , but won’t be able to do that until later since I’m on mobile at the moment and can’t backup the device names and automation linkings first.

**Reply by ziptbm** [reply]
> Also make sure that the SOMA Connect block on your Devices screen has the "Provided by custom integration" icon and that the documentation links point here. I notice you can enable/disable debug logging via that screen too. The frontend developers keep adding buttons I never notice. :)

Yes, that’s where I also enabled debug logging. It links here.

**Reply by Djelibeybi** [reply]
Maybe try completely rebooting your Home Assistant device? Can't think of anything else.

**Reply by ziptbm** [reply]
Should the config yaml contain custom_components.soma_connect?

I think it currently has only custom_components.soma since that what the text I copied from the HA forums thread.

**Reply by Djelibeybi** [reply]
Whoops, you're right. The logger configuration should use `custom_components.soma_connect`. Sorry abou that!

**Reply by ziptbm** [reply]
Thanks.  Changed it, rebooted, but still nothing.   Trying a complete system reboot.  

Or maybe the debug option in Ui is messing it up?  Will try that next if this doesn’t work.

**Comment by Djelibeybi** [comment]
I've updated the forum thread too, just in case anyone else tries it.

**Reply by ziptbm** [reply]
I just emailed over some of the logs.  Please let me know if that's helpful.  Thanks!

**Reply by Djelibeybi** [reply]
The Smart Shades app actively polls each motor, so it sees live data, while SOMA Connect uses passive bluetooth and listens for updates from each motor. So, you need to give both the SOMA Connect device and the integration a lot more time to update, I think.

 My integration only checks every 10 seconds, for example. And it can take SOMA Connect over 30 seconds to update its view of the world.

**Reply by Djelibeybi** [reply]
The logs indicate that SOMA Connect hasn't noticed the shades have stopped before reaching their target position.

**Reply by ziptbm** [reply]
Ok, I was trying to do a quick test to get some logs tonight, but clearly that isn't a good sample.  I'll do this again tomorrow morning (when folks are sleeping) and give it more time, and then provide those logs.   I usually wake up to find that someone in my home has opened them, but effectively leaves them in the stuck state, so they sit for 30-60 minutes at that point before I "fix" them manually.   When that happens, Smart Shades shows the correct position, but HA still does not.  Let's see if that reproduces tomorrow.

**Reply by ziptbm** [reply]
Strike that - i just did another test and allowed it to sit longer (roughly 3 min) and still not see the correct reading in HA.   Those logs with notes were just emailed over.

**Reply by Djelibeybi** [reply]
Your SOMA Connect is not updating the position, so there's not much I can do unfortunately. I'll see if there's a way to force it to update, but I'm not optimistic.

**Reply by ziptbm** [reply]
Do you think there's a problem with my SOMA Connect?  I'll go check it for firmware updates.

I have it hardwired into my network and line of sight (3-10 meters away) from each Soma device.

**Reply by Djelibeybi** [reply]
Definitely worth checking to make sure you have the latest firmware

**Reply by ziptbm** [reply]
It's the Raspi version on 2.3.1, so it looks like it's on the latest version.

**Reply by ziptbm** [reply]
Just curious @Djelibeybi , does your SOMA Connect behave differently?  Which device are you using?

**Reply by Djelibeybi** [reply]
Sorry, I missed this reply yesterday: my shades don't stop so I'm having a hard time reproducing your issue.

**Reply by ziptbm** [reply]
ok!  What Soma Connect do you use?

I've submitted a support ticket to Soma to see if there's anything else I can do to improve the reporting information from Soma Connect since the Smart Shades app is showing the correct value.  I really don't know what else to do here...

**Reply by Djelibeybi** [reply]
I use a Raspberry Pi 4 running SOMA Connect 2.3.1. There is a new USB SOMA Connect device that may be better, but I've not tried it myself yet.

**Reply by ziptbm** [reply]
Interesting. I’m on the original pi-based model purchased from SOMA.  We will see what they say to my support request then. 

In the meantime, should I revert off your beta build?

Or is there anything else I can do or test on this one?

**Reply by Djelibeybi** [reply]
This beta should be more reliable overall than the built-in integration as it has actual retries and reconnect logic that the built-in version does not. I would also suggest increasing the speed of your blinds if they keep stopping, as usually its caused by the speed being too low for the motor to actually move the blind.

**Reply by ziptbm** [reply]
Ok, I’ll keep the beta for now.

thanks, I keep them all at 100% and just double checked to make sure.  

If I don’t get anywhere with SOMA support, I may just have to try reinstalling.  However, this only appears to happen in colder months….

**Comment by ziptbm** [comment]
I just noticed when a shade does not complete the open request, if I tap/click on the stop button in HA for that shade, the location updates correctly.

maybe it thinks it’s still opening and therefore not updating?  I’m mentioning in case there is something there can force the location update.  Or this might be my automation workaround. The problem is there are no signals to HA that it didn’t complete the request….at least in the UI.  

Soma support has been unhelpful so far.

**Reply by Djelibeybi** [reply]
The integration has no knowledge of how open or closed a shade is unti it asks the SOMA Connect. When you tell it to open/close or use the slider, it just sends a single command to SOMA Connect that says, "Adjust the shade to this"

My advice would be to time how long a successful opening and closing takes, then use that as a delay in your automation to then "stop" the shade.

**Comment by ziptbm** [comment]
FYI - I've adjusted my install, which appears to have helped a little - now the devices will just stop, but issuing another open command seems to successfully open them, so I don't have to close them a few percentage first.

Aside from that, SOMA support released a new firmware 2.3.3 in response to my case.  Per support, this update should do the following: "The position should update 30 seconds after the motor realizes that it's not moving. After 30 seconds of staying still the motor will stop trying to move and update the position."

I'll be testing this over the next few days.  Sharing the latest from SOMA in case this has any implications for the beta build I'm still running.

**Comment by Djelibeybi** [comment]
The Beta should notice the motor has stopped at most about 30 seconds after SOMA Connect does. Hopefully.

**Comment by ziptbm** [comment]
Two quick observations from the beta build following Soma's release of firmware 2.3.3.  It could be coincidental and of no consequence, but I'm sharing in case it's helpful.  The firmware update has significantly improved the position reporting back to HA in combination with this build.

1. **One of my devices does not know the model for some reason in HA and shows it as "unknown unknown."   Operation wise, I've seen no other issues.**
<img width="899" alt="image" src="https://user-images.githubusercontent.com/46136563/227955801-0f21caaa-5db0-4ff8-a207-f635714a3488.png">

2. **Each device in HA shows firmware 2.3.1, even though Smart Shades shows 2.3.3.**
<img width="896" alt="image" src="https://user-images.githubusercontent.com/46136563/227955936-8d10bc2b-f539-448c-9684-4b2077472f53.png">

**Reply by Djelibeybi** [reply]
Soma don't appear to have released the firmware publicly yet, and I can't reproduce or test things until they do. 

I've seen the unknown model problem with the current version too, it's just what SOMA Connect reports. The version may be similar. Probably why they haven't released it publicly yet. :)

**Comment by ziptbm** [comment]
Well, I’m not sure what’s going on here but I discovered one of my shades now reporting as open, but it’s physically closed and has been for over 12 hours.  The smart shades app shows as closed, but HA is showing closed.  Even after reloading the app or asking it to close, it still shows as open.  I can grab logs tomorrow, but here’s a screenshot of what it shows from the history on this shade. 

![A6894B24-0B72-4381-87F1-D0EBC92AA6B4](https://user-images.githubusercontent.com/46136563/228129273-82f1d0de-8b1c-4b64-861a-966798a1be42.png)

I’ll submit to soma too since this may be related to the new firmware.

**Comment by ziptbm** [comment]
Could the beta build be pinging the devices so much that it impacts the battery?

I've noticed three of the shades are losing battery significantly overnight as well, despite no movement or activity.  Soma has indicated that earlier 2S devices had a hardware defect, so they may be sending me replacements....not sure if related, but I'm starting to think there's a bigger issue here as each step to workaround is spurring a new "issue."

**Comment by Djelibeybi** [comment]
The beta doesn't ping the devices at all, it only talks to SOMA Connect. I have 5 motors in my house and the battery lasts weeks for each of them.

**Reply by ziptbm** [reply]
Thanks for confirming.  I think there’s an issue with my units then, so I’ll push for replacement to see if that helps.
